### PR TITLE
Extract createWaitToBeSuccess and createWaitToBeSuccess into base class

### PR DESCRIPTION
**Pull request by rzadp**

### CODE REVIEW

File: universal-login-sdk/lib/api/FutureWalletFactory.ts [hunk 1]
```diff
@@ -99,7 +79,10 @@ export class FutureWalletFactory extends MineableFactory {
 
       const deployment: Deployment = {
         waitForTransactionHash: this.createWaitForTransactionHash(deploymentHash),
-        waitToBeSuccess: this.createWaitToBeSuccess(deploymentHash, deployedWallet),
+        waitToBeSuccess: async () => {
+          await this.createWaitToBeSuccess(deploymentHash)();
+          return deployedWallet;
```
Comment (cehali) [inline universal-login-sdk/lib/api/FutureWalletFactory.ts]: Do we want here to return deployedWallet or status?

Comment (xitronix) [reply universal-login-sdk/lib/api/FutureWalletFactory.ts]: As I remember waitToBeSuccess should return deployedWallet and optionally we maybe will add `waitToBeMined` which will return status

Comment (cehali) [reply universal-login-sdk/lib/api/FutureWalletFactory.ts]: Ok yes, my mistake

File: universal-login-sdk/lib/core/services/MineableFactory.ts [hunk 2]
```diff
@@ -21,4 +23,27 @@ export class MineableFactory {
     }
     return false;
   }
+
+  protected createWaitForTransactionHash(hash: string) {
+    return async () => {
+      const predicate = (status: MineableStatus) => !this.hasTransactionHash(status);
+      const status: MineableStatus = await this.waitForStatus(hash, predicate);
```
[the remark this answers is not in the document]
Comment (rzadp) [reply universal-login-sdk/lib/core/services/MineableFactory.ts]: Done

File: universal-login-sdk/lib/core/services/MineableFactory.ts [hunk 3]
```diff
@@ -21,4 +23,27 @@ export class MineableFactory {
...
+      return status;
+    };
+  }
+
+  protected createWaitToBeSuccess(hash: string) {
+    return async () => {
+      const predicate = (status: MineableStatus) => !this.isMined(status.state);
+      const status: MineableStatus = await this.waitForStatus(hash, predicate);
```
[the remark this answers is not in the document]
Comment (rzadp) [reply universal-login-sdk/lib/core/services/MineableFactory.ts]: Done

**Pull request merged**